src/main/java/com/product/exception/NotFoundExceptionController.java

```diff
@@ -0,0 +1,16 @@
+package com.product.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+class NotFoundExceptionController {
```

Comment:
This should be general name.. not specific exception controller name..

As there will be other exceptions that you will throw in the same service, and all of them will be handled in the same ControllerAdvise class.

Reply:
I have changed name to ProductExceptionController as it will handle the exception at the service level.